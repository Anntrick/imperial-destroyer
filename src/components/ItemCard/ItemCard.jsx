import './ItemCard.scss'

export const ItemCard = (props) => {
  const img = props.data.terrain ? `/planets/${props.data.name.split(" ").join("").toLowerCase()}.png` : `/starships/${props.data.name.split(" ").join("").toLowerCase()}.png`

  const imgError = (error) => {
    console.log(error)
    error.target.src = props.errorImg; 

  }

  return (
    <div className="ItemCard">
      <img src={img} alt={props.data.name} onError={imgError}/>
      <h2>{props.data.name}</h2>

      {props.data.terrain &&
        <>
          <p>{props.data.terrain}</p>
          <p>{props.data.population == "unknown" ? "Unknown population" : `Population of ${props.data.population}`} </p>
        </>
      }

      {props.data.crew &&
        <>
          <p>Crew: {props.data.crew}</p>
          <p>Cargo capacity: {props.data.cargo_capacity}</p>
          <p>Cost: {props.data.cost_in_credits} Credits</p>
          <p>Passengers: {props.data.passengers} People</p>

        </>
      }

    </div>

  )

}