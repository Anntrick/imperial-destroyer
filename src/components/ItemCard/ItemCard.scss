.ItemCard {
    border: 1px solid white;
    margin: 1em;
    padding: 1em;
    width: fit-content;
    color: white;

    img {
        width: 100%;
        max-width: 20em;

    }
}