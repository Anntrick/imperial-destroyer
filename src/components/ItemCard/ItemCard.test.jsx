import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { ItemCard } from './ItemCard'


describe('ItemCard component', () => {
    it('renders correctly', () => {
        const data = {
            name: 'Tatooine',
            terrain: 'Desert',
            population: '200000',
        }
        const errorImg = '/error.png'
        const { container } = render(
            <ItemCard data={data} errorImg={errorImg} />
        )
        expect(container).toMatchSnapshot()
    })

    it('displays the correct image for planets', () => {
        const data = {
            name: 'Tatooine',
            terrain: 'Desert',
            population: '200000',
        }
        const errorImg = '/error.png'
        const { getByAltText } = render(
            <ItemCard data={data} errorImg={errorImg} />
        )
        const img = getByAltText('Tatooine')
        expect(img.getAttribute('src')).toBe('/planets/tatooine.png')
    })

    it('displays the correct image for starships', () => {
        const data = {
            name: 'Millennium Falcon',
            crew: '6',
            cargo_capacity: '100000',
            cost_in_credits: '100000',
            passengers: '6',
        }
        const errorImg = '/error.png'
        const { getByAltText } = render(
            <ItemCard data={data} errorImg={errorImg} />
        )
        const img = getByAltText('Millennium Falcon')
        expect(img.getAttribute('src')).toBe('/starships/millenniumfalcon.png')
    })

    it('displays the error image if the image URL is invalid', () => {
        const data = {
            name: 'Tatooine',
            terrain: 'Desert',
            population: '200000',
        }
        const errorImg = '/error.png'
        const { getByAltText } = render(
            <ItemCard data={data} errorImg={errorImg} />
        )
        const img = getByAltText('Tatooine')
        fireEvent.error(img)
        expect(img.getAttribute('src')).toBe(errorImg)
    })
})


