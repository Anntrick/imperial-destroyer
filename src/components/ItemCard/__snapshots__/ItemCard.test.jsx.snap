// Vitest Snapshot v1

exports[`ItemCard component > renders correctly 1`] = `
<div>
  <div
    class="ItemCard"
  >
    <img
      alt="Tatooine"
      src="/planets/tatooine.png"
    />
    <h2>
      Tatooine
    </h2>
    <p>
      Desert
    </p>
    <p>
      Population of 200000
       
    </p>
  </div>
</div>
`;
